crates/gpui/src/window.rs

```diff
@@ -3620,28 +3620,30 @@ impl Window {
             currently_pending.keystrokes = match_result.pending;
             currently_pending.focus = self.focus;
             currently_pending.timer = Some(self.spawn(cx, async move |cx| {
-                cx.background_executor.timer(Duration::from_secs(1)).await;
-                cx.update(move |window, cx| {
-                    let Some(currently_pending) = window
-                        .pending_input
-                        .take()
-                        .filter(|pending| pending.focus == window.focus)
-                    else {
-                        return;
-                    };
-
-                    let node_id = window.focus_node_id_in_rendered_frame(window.focus);
-                    let dispatch_path = window.rendered_frame.dispatch_tree.dispatch_path(node_id);
-
-                    let to_replay = window
-                        .rendered_frame
-                        .dispatch_tree
-                        .flush_dispatch(currently_pending.keystrokes, &dispatch_path);
-
-                    window.pending_input_changed(cx);
-                    window.replay_pending_input(to_replay, cx)
-                })
-                .log_err();
+                // TODO: This resets the current pending inputs after 1s. This doesn't really work with a which-key system.
```

Comment:
Looking for feedback for what to do with this. which-key is not usable if the pending keys are cleared after 1s.

Reply:
As per discuss below, I will implement that this is disable if the which-key menu is open.